### PR TITLE
[REF] bus, *: shortcut to send message to user via bus

### PR DESCRIPTION
**Pull request by seb-odoo**
\* = account, auth_signup, base_geolocalize, bus, calendar, crm_livechat, iap_mail, im_livechat, mail, point_of_sale, website_crm_partner_assign, website_livechat

Make a safe and simpler helper for usages of `<bus.bus>._sendone`. If the code is using this helper, it doesn't require any particular attention regarding security as the channel crafted is safe by design. It also reduces the possibility to have a channel that doesn't exist.

However supported channels still have to be added manually in `_build_bus_channel_list` depending on each user/guest needs and access rights.

Implementation:

A new mixin (`bus.listener.mixin`) implementing `_bus_send` can be added on models, which, when called, will send a message to listeners of these records.

In order to do that, models inheriting that mixin will need to also implement `_bus_channel`, returning the channel to use for the current record (by default it is the record itself).

The opportunity is taken to add various sending helpers that are used often, in particular `_bus_send_store`.

Transient message helper is also cleaned to use a similar pattern, to enforce markup usage, and to follow proper translation guidelines.

Co-Authored-By: Sébastien Theys <seb@odoo.com>
Co-Authored-By: william-andre <wan@odoo.com>

https://github.com/odoo/enterprise/pull/67733

Taken from https://github.com/odoo/odoo/pull/161495

### CODE REVIEW

**Review by kmagusiak — APPROVED** [review]

File: addons/im_livechat/models/im_livechat_channel.py [hunk 1]
```diff
@@ -103,23 +102,15 @@ def _compute_nbr_channel(self):
     def action_join(self):
         self.ensure_one()
         self.user_ids = [Command.link(self.env.user.id)]
-        self.env["bus.bus"]._sendone(
-            self.env.user.partner_id,
-            "mail.record/insert",
-            Store(
-                "LivechatChannel", {"id": self.id, "name": self.name, "hasSelfAsMember": True}
-            ).get_result(),
+        self.user._bus_send_store(
```
Comment (tsm-odoo) [inline addons/im_livechat/models/im_livechat_channel.py:105]:
```suggestion
        self.env.user._bus_send_store(
```

Comment (seb-odoo) [reply addons/im_livechat/models/im_livechat_channel.py:105]: Good catch, I checked 3 times and still missed that one.

**Review by alexkuhn — APPROVED** [review]
@robodoo delegate+

**Comment by seb-odoo** [comment]
@robodoo override=ci/security

In bus mixin: sending on safe channel (result of `_bus_channel()` which is enforced to be a record)

In pos mixin: sending on channel containing access token

To consider for a follow up PR: only allow sending on records from all places, and use token only to validate access to records in `_build_bus_channel_list`

**Comment by seb-odoo** [comment]
@robodoo r+

**Pull request closed without merging**